pennylane/ops/op_math/sprod.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the SProd class which contains logic for
+computing the scalar product of operations.
+"""
+from pennylane import math
+from .symbolicop import SymbolicOp
+
+
+def s_prod(scalar, operator):
+    """Associate an operator with a scalar to represent scalar multiplication."""
+    return SProd(scalar, operator)
+
+
+def _sprod(mat, scalar, dtype=None, cast_like=None):
```

Comment:
Still wondering about the need for the `dtype` and `cast_like` arguments.

Also, is there a reason we can't just do `scalar * mat`?

Reply:
To be honest, I didn't try that. I just assumed it would be safer to dispatch to autoray and let it figure what backend / type the resultant matrix should have. I would imagine that this would be required when the scalar provided is given in a different backend than the matrix / operator.

Reply:
Should I try the `scalar * mat` and keep it if it passes all tests?